crates/codegen/schema/src/types/precedence_parser.rs

```diff
@@ -0,0 +1,51 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use super::parser::ParserDefinition;
+
+#[allow(dead_code)]
+pub type PrecedenceParserRef = std::rc::Rc<PrecedenceParser>;
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields)]
+pub struct PrecedenceParser {
+    #[serde(default)]
+    pub name: Option<String>,
+    #[serde(flatten)]
+    pub definition: PrecedenceParserDefinition,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct PrecedenceParserDefinition {
+    #[schemars(title = "Operator Definitions")]
+    pub operators: Vec<OperatorDefinition>,
+
+    #[schemars(title = "Primary Expression References")]
+    pub primary_expressions: Vec<Reference>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct OperatorDefinition {
+    pub name: String,
+    pub model: OperatorModel,
+    #[serde(flatten)]
+    pub definition: ParserDefinition,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, JsonSchema, PartialEq, Eq, Hash)]
+#[serde(deny_unknown_fields)]
+pub enum OperatorModel {
+    BinaryLeftAssociative,
+    BinaryRightAssociative,
+    UnaryPrefix,
+    UnarySuffix,
```

Comment:
Thoughts on using `suffix` vs `postfix`?

In my mind, a suffix is a part of the atom itself, while a postfix is something that comes after the atom.

Some online threads/articles explaining the semantic difference I found (not authoritative):

- https://grammarhow.com/postfix-vs-suffix/
- https://english.stackexchange.com/a/81293

> ‘Postfix’ and ‘suffix’ are two entirely different terms with entirely different meanings. We use ‘suffix’ in linguistics to signify attaching an affix at the end of a root word, while we use ‘postfix’ in computer and programming which signifies a notation that puts the operands before the operator.